### PR TITLE
nGraph as device

### PR DESCRIPTION
**Pull request by avijit-nervana**
Experimental work to connect nGraph as a TensorFlow device. To use this mode use the following environment variable: `NGTF_USE_DEVICE`. Currently only the simplest network will work as there are only three ops supported - const, add, mul.

### CODE REVIEW

File: experimental/ngraph_device/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# nGraph As a Device
+
+This is an experimental project to connect nGraph with TensorFlow as a TensorFlow device (such as a CPU or a GPU)
+Currently nGraph is added an optimizer for the CPU in which computation nodes are replaced by one or more nGraph 
+encapsulatr nodes. However, transferring the tensors between nGraph device and the host CPU involves several 
+layers of book keeping.
+
+In this experimental project, nGraph will be registered as NGRAPH device and will take dvantages of the
```
Comment (sayantan-nervana) [inline experimental/ngraph_device/README.md]:
```suggestion
In this experimental project, nGraph will be registered as NGRAPH device and will take advantages of the 
```

File: experimental/ngraph_device/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+This is an experimental project to connect nGraph with TensorFlow as a TensorFlow device (such as a CPU or a GPU)
+Currently nGraph is added an optimizer for the CPU in which computation nodes are replaced by one or more nGraph 
+encapsulatr nodes. However, transferring the tensors between nGraph device and the host CPU involves several 
+layers of book keeping.
+
+In this experimental project, nGraph will be registered as NGRAPH device and will take dvantages of the 
+TensorFlow placer. 
+
+Earlier versionb of the nGraph bridge used the device registration mechanism. Those changes
```
Comment (sayantan-nervana) [inline experimental/ngraph_device/README.md]:
```suggestion
Earlier versions of the nGraph bridge used the device registration mechanism. Those changes
```

Comment (sayantan-nervana) [reply experimental/ngraph_device/README.md]: We could add the "`NGTF_USE_DEVICE=1` flag to enable this feature" information in the README

File: experimental/ngraph_device/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# nGraph As a Device
+
+This is an experimental project to connect nGraph with TensorFlow as a TensorFlow device (such as a CPU or a GPU)
+Currently nGraph is added an optimizer for the CPU in which computation nodes are replaced by one or more nGraph 
+encapsulatr nodes. However, transferring the tensors between nGraph device and the host CPU involves several layers of book keeping.
```
Comment (kanvi-nervana) [inline experimental/ngraph_device/README.md]: typo `encapsulatr` ---> encapsulate

File: experimental/ngraph_device/device_test.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# The following filtering will supress all the `FutureWarning` from numpy
```
Comment (kanvi-nervana) [inline experimental/ngraph_device/device_test.py]: missing License header

**Pull request merged**